### PR TITLE
DDF-3622 Adds admin config for list templates

### PR DESCRIPTION
**Pull request by andrewkfiedler**
Update of https://github.com/codice/ddf/pull/2960 since it was way out of date.

#### What does this PR do?
 - Allows the admin to specify commonly created lists for users to utilize.

#### Who is reviewing it? 
@bdeining
@millerw8
@djblue 
@mackncheesiest 

#### How should this be tested? (List steps with links to updated documentation)
Go to the admin UI and add a new template like:
`{"id":"Pizza Deliveries", "list.icon": "tasks", "list.cql": "(\"anyText\" ILIKE 'pizza')"}`

#### What are the relevant tickets?
[3622](https://codice.atlassian.net/browse/DDF-3622)

#### Screenshots (if appropriate)
screen shot 2018-02-13 at 4 51 40 pm
screen shot 2018-02-13 at 4 52 41 pm
screen shot 2018-02-13 at 4 52 48 pm
screen shot 2018-02-13 at 4 53 02 pm

### CODE REVIEW

File: ui/packages/catalog-ui-search/src/main/webapp/component/list-editor/list-editor.view.js [hunk 1]
```diff
@@ -55,6 +59,37 @@ module.exports = Marionette.LayoutView.extend({
...
+              value: template.id,
+              class: List.getIconMapping()[template['list.icon']]
+            })
+          )),
+          id: 'Template'
+      });
+      this.listTemplate.show(new PropertyView({
+          model: propertyModel
+      }));
+      this.listTemplate.currentView.turnOnLimitedWidth();
```
Comment (mackncheesiest) [inline ui/packages/catalog-ui-search/src/main/webapp/component/list-editor/list-editor.view.js]: ✏️ Not needed anymore, right?

**Comment by andrewkfiedler** [comment]
Did a hero to make sure everything still worked.  Tested adding a few templates, and then used them in the UI and verified they got the correct icon and the correct filters.  Also tested templates that don't parse correctly and the UI handled them gracefully.

**Comment by andrewkfiedler** [comment]
build now

**Comment by bdeining** [comment]
successfully hero'd on chrome and firefox

**Pull request merged**